StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentMethodFormViewController.swift

```diff
@@ -24,14 +24,15 @@ class PaymentMethodFormViewController: UIViewController {
     let analyticsHelper: PaymentSheetAnalyticsHelper
     weak var delegate: PaymentMethodFormViewControllerDelegate?
     var paymentOption: PaymentOption? {
-        // TODO Copied from AddPaymentMethodViewController but this seems wrong; we shouldn't have a divergent path for link. Where is the setDefaultBillingDetailsIfNecessary call, for example?
-        if let linkEnabledElement = form as? LinkEnabledPaymentMethodElement {
-            return linkEnabledElement.makePaymentOption(intent: intent, elementsSession: elementsSession)
-        }
 
         let params = IntentConfirmParams(type: paymentMethodType)
         params.setDefaultBillingDetailsIfNecessary(for: configuration)
+
         if let params = form.updateParams(params: params) {
+            if let linkEnabledElement = form.getAllUnwrappedSubElements().compactMap({ $0 as? LinkEnabledPaymentMethodElement }).first {
+                return linkEnabledElement.makePaymentOption(intentConfirmParams: params)
+            }
+
```

Comment:
Not sure if this is correct -- we were previously short circuiting setting default billing details if needed.  Seems like this was not intentional per the comment that I erased.